tools/java/runfiles/testing/AutoBazelRepositoryTest.java

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.runfiles;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Unit tests for {@link AutoBazelRepository}.
+ */
+@RunWith(JUnit4.class)
+@AutoBazelRepository
+public final class AutoBazelRepositoryTest {
```

Comment:
1. This looks like a weird tests, the arrange part is happening by the annotation. I'd be more comfortable the test is moved into the shell test.

2. Why isn't repository name properly injected? This is a java_test and it is in a repository. Shouldn't this be a failure?

Reply:
Agreed.